npm/webpack-dev-server/src/makeDefaultWebpackConfig.ts

```diff
@@ -24,9 +24,17 @@ export function makeDefaultWebpackConfig (
 
   debug(`Using HtmlWebpackPlugin version ${version} from ${importPath}`)
 
+  // Webpack 5 uses `emitOnErrors: true`
+  // Webpack <= 4 uses `noEmitOnErrors: false`
+  const emitOnErrorsProperty = {
+    [config.sourceWebpackModulesResult.webpack.majorVersion === 5 ? 'emitOnErrors' : 'noEmitOnErrors']:
+    config.sourceWebpackModulesResult.webpack.majorVersion === 5,
+  }
+
   const finalConfig = {
     mode: 'development',
     optimization: {
+      ...emitOnErrorsProperty,
```

Comment:
```suggestion
  const optimization = <Record<string, any>>{}

  if (config.sourceWebpackModulesResult.webpack.majorVersion === 5) {
    optimization.emitOnErrors = true
  } else {
    optimization.noEmitOnErrors = false
  }

  const finalConfig = {
    mode: 'development',
    optimization: {
      ...optimization,
```

I found this a bit hard to parse, being more explicit reads a bit nicer.